File: app/src/main/java/com/mns/ghassedak/ESP8266Connector.java
package com.mns.ghassedak;

import android.content.Context;
import android.util.Log;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

public class ESP8266Connector {

    private Context ctx;
    private String root;
    private RequestQueue mRequestQueue;

    public ESP8266Connector(Context context, String ip, String port) {
        this.ctx = context;
        this.root = "http://" + ip + ":" + port;
    }

    public void moveForward() {
        sendRequest("N");
    }

    public void moveBackward() {
        sendRequest("S");
    }

    public void turnRight() {
        sendRequest("E");
    }

    public void turnLeft() {
        sendRequest("W");
    }

    public void stopMoving() {
        sendRequest("stop");
    }

    public void moveNE() {
        sendRequest("NE");
    }

    public void moveNW() {
        sendRequest("NW");
    }

    public void moveSE() {
        sendRequest("SE");
    }

    public void moveSW() {
        sendRequest("SW");
    }

    public void sendVip() {
        sendRequest("Vip");
    }


    private void sendRequest(String stateValue) {
        String request = this.root + "/?State=" + stateValue;
        StringRequest stringRequest = new StringRequest(Request.Method.GET, request,
                new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {
                        Log.i("send Request Response", response);
                    }
                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Log.i("send Request Error", error.toString());
            }
        });
        stringRequest.setShouldCache(false);
        getRequestQueue().add(stringRequest);
    }

    private RequestQueue getRequestQueue() {
        if (mRequestQueue == null) {
            mRequestQueue = Volley.newRequestQueue(ctx);
        }
        return mRequestQueue;
    }

    public void clearRequestQueue() {
        getRequestQueue().cancelAll(new RequestQueue.RequestFilter() {
            @Override
            public boolean apply(Request<?> request) {
                return true;
            }
        });
    }

}

File: app/src/main/java/com/mns/ghassedak/ESP8266Connector.kt
package com.mns.ghassedak

import android.content.Context
import android.util.Log
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley

class ESP8266Connector(private val ctx: Context, ip: String, port: String) {
    private val root: String = "http://$ip:$port"
    private var mRequestQueue: RequestQueue? = null
    fun moveForward() {
        sendRequest("N")
    }

    fun moveBackward() {
        sendRequest("S")
    }

    fun turnRight() {
        sendRequest("E")
    }

    fun turnLeft() {
        sendRequest("W")
    }

    fun stopMoving() {
        sendRequest("stop")
    }

    fun moveNE() {
        sendRequest("NE")
    }

    fun moveNW() {
        sendRequest("NW")
    }

    fun moveSE() {
        sendRequest("SE")
    }

    fun moveSW() {
        sendRequest("SW")
    }

    fun sendVip() {
        sendRequest("Vip")
    }

    fun setSound() {
        sendRequest("sound")
    }

    private fun sendRequest(stateValue: String) {
        val request = this.root + "/?State=" + stateValue
        val stringRequest = StringRequest(
            Request.Method.GET, request,
            { response ->
                Log.i(
                    "send Request Response",
                    response!!
                )
            }) { error -> Log.i("send Request Error", error.toString()) }
        stringRequest.setShouldCache(false)
        requestQueue.add(stringRequest)
    }

    private val requestQueue: RequestQueue
        get() {
            if (mRequestQueue == null) {
                mRequestQueue = Volley.newRequestQueue(ctx)
            }
            return mRequestQueue!!
        }

    fun clearRequestQueue() {
        requestQueue.cancelAll { true }
    }

}